ddtrace/_trace/apm_filter.py

```diff
@@ -0,0 +1,22 @@
+import os
+from typing import List
+from typing import Optional
+
+from ddtrace._trace.processor import TraceProcessor
+from ddtrace._trace.span import Span
+from ddtrace.internal.utils.formats import asbool
+
+
+class APMTracingEnabledFilter(TraceProcessor):
+    """
+    Trace processor that drops all APM traces when DD_APM_TRACING_ENABLED is set to a falsy value.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._apm_tracing_enabled = asbool(os.getenv("DD_APM_TRACING_ENABLED", "true"))
```

Comment:
Can we use the get_config defined in ddtrace.internal.telemetry here? This will allow us to use file based configs here (aka stable config)

Reply:
We could also make this the first user processor to avoid connecting profiles to resource names that won't exist

Reply:
hmm i'm not sure how well established this flag is so a little hesitant to add it to a common code path here